src/sentry/static/sentry/app/utils/discover/fieldRenderers.tsx

```diff
@@ -198,10 +199,18 @@ const SPECIAL_FIELDS: SpecialFields = {
   project: {
     sortField: 'project',
     renderFunc: (data, {organization}) => {
-      const project = organization.projects.find(p => p.slug === data.project);
       return (
         <Container>
-          {project ? <ProjectBadge project={project} avatarSize={16} /> : data.project}
+          <Projects orgId={organization.slug} slugs={[data.project]}>
+            {({projects}) => {
+              const project = projects.find(p => p.slug === data.project);
+              return project ? (
+                <ProjectBadge project={project} avatarSize={16} />
+              ) : (
+                data.project
+              );
```

Comment:
Should we always render project badge? 

```suggestion
              return <ProjectBadge project={project ? project : {slug: data.project}} avatarSize={16} />
```

Aside, there's a prop to passthrough slug as a placeholder: https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/utils/projects.tsx#L85 so this should always find a project and have a placeholder. (Which I think would make a better experience).